### PR TITLE
Implement JSON.parseImmutable

### PR DESCRIPTION
**Pull request by nicolo-ribaudo**
I'm marking this as a draft PR because there are still some undecided steps (https://github.com/tc39/proposal-record-tuple/issues/120), so we first need to write the spec text

### CODE REVIEW

File: packages/record-tuple-polyfill/src/index.js [hunk 1]
```diff
@@ -16,3 +16,4 @@
 
 export { Record } from "./record";
 export { Tuple } from "./tuple";
+export { parseImmutable } from "./json";
```
Comment (nicolo-ribaudo) [inline packages/record-tuple-polyfill/src/index.js:19]: Is it ok if I export it like this instead of mutating the JSON object?

Comment (littledan) [reply packages/record-tuple-polyfill/src/index.js:19]: Yes, I think that's preferable.

**Comment by rricard** [comment]
You need to rebase this one to use the new fake json namespace you created

**Comment by littledan** [comment]
I merged the spec part since I think it's plenty stable.

**Pull request merged**